app/src/main/java/org/mozilla/tv/firefox/pocket/PocketEndpoint.kt

```diff
@@ -8,24 +8,47 @@ import android.net.Uri
 import android.support.annotation.AnyThread
 import android.support.annotation.VisibleForTesting
 import android.util.Log
+import io.reactivex.Maybe
+import io.reactivex.Single
+import io.reactivex.schedulers.Schedulers
+import kotlinx.coroutines.runBlocking
 import okhttp3.Request
 import org.json.JSONException
 import org.json.JSONObject
 import org.mozilla.tv.firefox.BuildConfig
 import org.mozilla.tv.firefox.ext.executeAndAwait
 import org.mozilla.tv.firefox.ext.flatMapObj
+import org.mozilla.tv.firefox.utils.Endpoint
 import org.mozilla.tv.firefox.utils.OkHttpWrapper
+import org.mozilla.tv.firefox.utils.Response
 import java.io.IOException
 
 private const val LOGTAG = "PocketEndpoint"
 
+typealias PocketData = List<PocketViewModel.FeedItem.Video>
+
 /**
  * Make requests to the Pocket endpoint and returns internal objects.
  *
  * The methods of this class call the endpoint directly and does not cache results or rate limit,
  * outside of the network layer (e.g. with OkHttp).
  */
-open class PocketEndpoint(private val appVersion: String, private val pocketEndpoint: Uri?) {
+open class PocketEndpoint(private val appVersion: String, private val pocketEndpoint: Uri?): Endpoint<PocketData> {
+
+    /**
+     * Wraps the suspend function [getRecommendedVideos] in a [Maybe]
+     */
+    override fun request(): Single<Response<PocketData>> {
+        fun requestAndWrap(): Single<Response<PocketData>> {
+            val videos = runBlocking { getRecommendedVideos() }
+            return when {
+                videos == null || videos.isEmpty() -> Single.just(Response.Failure())
+                else -> Single.just(Response.Success(videos))
+            }
+        }
+        return requestAndWrap()
+            .subscribeOn(Schedulers.io())
```

Comment:
I find it obfuscating to put my scheduling inside of the observable source instead of as an operator when subscribing; personal preferences on this vary, but IMO it should be a project-level decision one way or the other.

Reply:
That's a good point, I'll move that.